### ISSUE TITLE
map-get flagged and variable for property / color

### ISSUE THREAD

**Issue by mbulfair** [sasstools/sass-lint#966]
What version of Sass Lint are you using?
**1.10.2**

Please include any relevant parts of your configuration
```
  no-color-hex: 0
  no-color-keywords: 2
  no-color-literals:
    - 2
    - allow-map-identifiers: true
    - allow-rgba: true
    - allow-variable-identifiers: true
  variable-for-property:
    - 2
    - properties:
        - color
        - font
```
What did you expect to happen?
_This is expected, but (see below)_
`color: map-get($uiDarkMap, display1Color);`
**Is there an exception for this function of map-get, if what's being passed is a variable? I want colors to have no hex values, but I want variables to be a hex value.**

What actually happened? Please include any error messages given to you by Sass Lint.

getting this error: 
`variable-for-property: Values for properties of type 'color' may only be variables`

**Comment by DanPurdy** [comment]
Related to #853 I reckon. I think an overhaul of this rule is required now. It's way too fiddly. My fault..

**Comment by DanPurdy** [comment]
Released in 1.12.0
